server/src/main.rs

```diff
@@ -107,35 +130,65 @@ impl LanguageServer for Backend {
         self.client
             .log_message(MessageType::INFO, "Slice language server config changed")
             .await;
-        let workspace_uri = self.workspace_uri.lock().await;
 
         // Update the sources directory if it has changed
-        let sources_directory = self
-            .get_sources_directory(&workspace_uri, Some(&params))
-            .await
-            .ok();
-
-        let reference_directories = self
-            .get_reference_directories(&workspace_uri, Some(&params))
-            .await
-            .ok();
+        let (sources_directory, reference_directories) = {
+            let workspace_uri = self.workspace_uri.lock().await;
+            let s = self
+                .get_sources_directory(&workspace_uri, Some(&params))
+                .await
+                .ok();
+
+            let r = self
+                .get_reference_directories(&workspace_uri, Some(&params))
+                .await
+                .ok();
+            (s, r)
+        };
 
         // Check if either sources or reference directories have changed
         if sources_directory.is_some() || reference_directories.is_some() {
```

Comment:
What is the configuration was set to something and is now None? Shouldn't we also update and recompile.